h2/src/main/org/h2/engine/Session.java

```diff
@@ -12,6 +12,7 @@
 import java.util.LinkedList;
 import java.util.Map;
 import java.util.Random;
+import java.util.Stack;
```

Comment:
Does it really make sense to use `Stack` here which has all `synchronized` methods?

Reply:
Changed to ArrayDeque - the collection package unsynch'd Stack